src/Planners/CMakeLists.txt

```diff
@@ -7,12 +7,39 @@ if (FRAMEWORK_COMPILE_Planners)
   set(H_PREFIX include/BipedalLocomotion/Planners)
 
   add_bipedal_locomotion_library(
-    NAME                   Planners
-    PUBLIC_HEADERS         ${H_PREFIX}/ConvexHullHelper.h ${H_PREFIX}/DCMPlanner.h ${H_PREFIX}/TimeVaryingDCMPlanner.h ${H_PREFIX}/QuinticSpline.h ${H_PREFIX}/SO3Planner.h ${H_PREFIX}/SO3Planner.tpp ${H_PREFIX}/SwingFootPlanner.h
-    SOURCES                src/ConvexHullHelper.cpp  src/DCMPlanner.cpp src/TimeVaryingDCMPlanner.cpp src/QuinticSpline.cpp src/SwingFootPlanner.cpp
-    PUBLIC_LINK_LIBRARIES  Eigen3::Eigen BipedalLocomotion::ParametersHandler BipedalLocomotion::System BipedalLocomotion::Contacts
-    PRIVATE_LINK_LIBRARIES Qhull::qhullcpp Qhull::qhullstatic_r casadi BipedalLocomotion::Math BipedalLocomotion::TextLogging
-    INSTALLATION_FOLDER    Planners)
+    NAME Planners
+    PUBLIC_HEADERS
+    ${H_PREFIX}/ConvexHullHelper.h
+    ${H_PREFIX}/DCMPlanner.h
+    ${H_PREFIX}/TimeVaryingDCMPlanner.h
+    ${H_PREFIX}/QuinticSpline.h
+    ${H_PREFIX}/SO3Planner.h
+    ${H_PREFIX}/SO3Planner.tpp
+    ${H_PREFIX}/SwingFootPlanner.h
+    ${H_PREFIX}/UnicyclePlanner.h
+    SOURCES
+    src/ConvexHullHelper.cpp
+    src/DCMPlanner.cpp
+    src/TimeVaryingDCMPlanner.cpp
+    src/QuinticSpline.cpp
+    src/SwingFootPlanner.cpp
+    src/UnicyclePlanner.cpp
+    PUBLIC_LINK_LIBRARIES
+    Eigen3::Eigen
+    BipedalLocomotion::ParametersHandler
+    BipedalLocomotion::System
+    BipedalLocomotion::Contacts
+    PRIVATE_LINK_LIBRARIES
+    Qhull::qhullcpp
+    Qhull::qhullstatic_r
+    casadi
+    UnicyclePlanner
+    iDynTree::idyntree-core
+    BipedalLocomotion::Math
+    BipedalLocomotion::TextLogging
+    INSTALLATION_FOLDER Planners)
+
+  target_compile_features(Planners PRIVATE cxx_std_17)
```

Comment:
This should not be here. It is already in https://github.com/dic-iit/bipedal-locomotion-framework/blob/master/cmake/AddBipedalLocomotionLibrary.cmake#L70. If the compilation fails without it, it is a problem

Reply:
Compilation fails on windows, `std::optional` cannot be found, no clue why. cc @traversaro

Reply:
I don't see a ``#include<optional>`` though :thinking: